packages/eas-cli/src/commands/update/index.ts

```diff
@@ -335,6 +333,10 @@ export default class UpdatePublish extends EasCommand {
           throw new Error('You must specify the update group to republish.');
         }
 
+        const { updates } = await ensureBranchExistsAsync({
+          appId: projectId,
+          name: branchName,
+        });
```

Comment:
Passing in the limit here would be a good idea because the number of updates to query from the history is part of this command's UX, rather than relying on the default limit `ensureBranchExistsAsync` happens to choose.

Also, on line 351 we maybe should change the error message to say, "There are no [recent] updates" since we now fetch only a subset of the updates.

Reply:
> Passing in the limit here would be a good idea because the number of updates to query from the history is part of this command's UX, rather than relying on the default limit ensureBranchExistsAsync happens to choose.

This *should* be addressed in the follow up [PR](https://github.com/expo/eas-cli/pull/1148) which deals solely with the republish flow.